npm/qsharp/ux/circuit-vis/panel.ts

```diff
@@ -0,0 +1,332 @@
+    metadata.type = gate === "X" ? GateType.Cnot : GateType.ControlledUnitary;
+    metadata.label = gate;
+    if (gate !== "X") {
+      metadata.targetsY = [[target]];
+    }
+  } else if (gate === "X") {
+    metadata.type = GateType.X;
+    metadata.label = gate;
+  } else {
+    metadata.type = GateType.Unitary;
+    metadata.label = gate;
+    metadata.targetsY = [[target]];
+  }
+
+  if (displayArgs != null) metadata.displayArgs = displayArgs;
+
+  metadata.width = getGateWidth(metadata);
+  metadata.x = x + 1 + metadata.width / 2; // offset by 1 for left padding
+
+  return metadata;
+};
+
+/**
+ * Generate an SVG gate element for the Toolbox panel based on the type of gate.
+ * This function retrieves the operation metadata from the gate dictionary,
+ * formats the gate, and returns the corresponding SVG element.
+ *
+ * @param gateDictionary - The dictionary containing gate operations.
+ * @param type - The type of gate. Example: 'H' or 'X'.
+ * @param x - The x coordinate at the starting point from the left.
+ * @param y - The y coordinate at the starting point from the top.
+ * @returns The generated SVG element representing the gate.
+ * @throws Will throw an error if the gate type is not available in the dictionary.
+ */
+const _gate = (
+  gateDictionary: GateDictionary,
+  type: string,
+  x: number,
+  y: number,
+): SVGElement => {
+  const operation = gateDictionary[type];
+  if (operation == null) throw new Error(`Gate ${type} not available`);
+  const metadata = toMetadata(operation, x, y);
+  metadata.dataAttributes = { type: type };
+  const gateElem = formatGate(metadata).cloneNode(true) as SVGElement;
+  gateElem.setAttribute("toolbox-item", "true");
+
+  return gateElem;
+};
+
+/**
+ * Interface for gate dictionary
+ */
+interface GateDictionary {
+  [index: string]: Operation;
+}
+
+/**
+ * Object for default gate dictionary
+ */
+const defaultGateDictionary: GateDictionary = {
+  RX: {
+    gate: "Rx",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  RY: {
+    gate: "Ry",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  RZ: {
+    gate: "Rz",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  X: {
+    gate: "X",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  Y: {
+    gate: "Y",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  Z: {
+    gate: "Z",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  H: {
+    gate: "H",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  S: {
+    gate: "S",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  T: {
+    gate: "T",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  Measure: {
+    gate: "Measure",
+    isMeasurement: true,
+    controls: [{ qId: 0, type: RegisterType.Qubit }],
+    targets: [{ qId: 0, type: RegisterType.Classical, cId: 0 }],
+  },
+  Reset: {
+    gate: "|0〉",
+    targets: [{ qId: 0, type: RegisterType.Qubit }],
+  },
+  ResetX: {
```

Comment:
I don't think `|1>` is ResetX. 0 and 1 are in the Z basis. I would have thought ResetX maps to `|+>`?

Reply:
Oh good point. `ResetX` does mean something else. I've renamed this to `ResetOne`.

Reply:
Actually, I realized last night I was wrong here. If a qubit starts in the `|0>` state, then applying an X does indeed put it into the `|1>` state. Applying an H puts it into what is commonly called the `|+>` state which is along the X basis of measurement, which I was getting confused with. Sorry!